pjsip/src/pjsip/sip_transaction.c

```diff
@@ -3093,10 +3093,14 @@ static pj_status_t tsx_on_state_proceeding_uac(pjsip_transaction *tsx,
         tsx_cancel_timer( tsx, &tsx->timeout_timer );
         unlock_timer(tsx);
 
-        /* For INVITE, the state moves to Terminated state (because ACK is
-         * handled in TU). For non-INVITE, state moves to Completed.
+        /* For INVITE and BYE, the state moves to Terminated state.
+         * For INVITE it is because ACK is handled in TU.
+         * For BYE it is because ACK should not be sent as response on 200 OK.
+         * For other methods, state moves to Completed.
          */
-        if (tsx->method.id == PJSIP_INVITE_METHOD) {
+        if (tsx->method.id == PJSIP_INVITE_METHOD ||
+            tsx->method.id == PJSIP_BYE_METHOD)
+        {
             tsx_set_state( tsx, PJSIP_TSX_STATE_TERMINATED, 
                            PJSIP_EVENT_RX_MSG, event->body.rx_msg.rdata, 0 );
             //return PJSIP_ETSXDESTROYED;
```

Comment:
According to the standard, non-INVITE tsx must go through the Completed state, I think the current flow is already correct.

https://www.rfc-editor.org/rfc/rfc3261#section-17.1.2:
```
   If a
   final response (status codes 200-699) is received while in the
   "Proceeding" state, the response MUST be passed to the TU, and the
   client transaction MUST transition to the "Completed" state.

   Once the client transaction enters the "Completed" state, it MUST set
   Timer K to fire in T4 seconds for unreliable transports, and zero
   seconds for reliable transports.  The "Completed" state exists to
   buffer any additional response retransmissions that may be received
   (which is why the client transaction remains there only for
   unreliable transports).
```

This patch works for the original "problem" reported in the PR because the dialog is destroyed immediately when BYE tsx is terminated immediately. But, terminating BYE tsx immediately (for unreliable transport) seems to violate the standard. Just curious whether the original "problem" is really an issue?

Reply:
The problem comes from a failed test case in the following test plan: [https://www.etsi.org/deliver/etsi_ts/102000_102099/10202702/04.01.01_60/ts_10202702v040101p.pdf](https://www.etsi.org/deliver/etsi_ts/102000_102099/10202702/04.01.01_60/ts_10202702v040101p.pdf) Test: SIP_CC_OE_CR_V_008.
It could be argued that the test is too strict and that in reality it should not cause any problems. But we still encountered problems and need a way to fix this.

Reply:
How about the snippet from the RFC that I quoted above:
```
Alice confirms receipt of
   the BYE with a 200 (OK) response, which terminates the session and
   the BYE transaction.
```

Reply:
> How about the snippet from the RFC that I quoted above:
> 
> ```
> Alice confirms receipt of
>    the BYE with a 200 (OK) response, which terminates the session and
>    the BYE transaction.
> ```

It looks like a sample scenario (which indeed can happen on a reliable transport) rather than rules. There are some other sample scenarios that the BYE transaction is not terminated immediately, but going through Completed state and waiting for timer K [here](https://www.rfc-editor.org/rfc/rfc5407.html).